sky/shell/shell.cc

```diff
@@ -68,6 +73,81 @@ void ServiceIsolateHook(bool running_precompiled) {
   }
 }
 
+class JSONResponseBuilder {
+ public:
+  void Append(const char* str) {
+    for (intptr_t i = 0; str[i] != '\0'; i++) {
+      output_.push_back(str[i]);
+    }
+  }
+
+  void Append(const std::string& str) {
+    for (size_t i = 0; i < str.size(); i++) {
+      output_.push_back(str.data()[i]);
+    }
+  }
+
+  void Append(uintptr_t num) {
+    Append(std::to_string(num));
+  }
+
+  std::string ToString() {
+    return std::string(output_.data(), output_.size());
+  }
+
+ private:
+  std::vector<char> output_;
+};
+
+// This callback will be invoked on a Dart worker thread.
+bool FlutterRunInView(const char* method,
+                      const char** param_keys,
+                      const char** param_values,
+                      intptr_t num_params,
+                      void* user_data,
+                      const char** json_object) {
+  // TODO(johnmccutchan): Implement this function.
+  *json_object = strdup("{\"type\": \"Success\"}");
+  return false;
+}
+
+// This callback will be invoked on a Dart worker thread.
+bool FlutterListViews(const char* method,
+                      const char** param_keys,
+                      const char** param_values,
+                      intptr_t num_params,
+                      void* user_data,
+                      const char** json_object) {
+  // Ask the Shell for the list of platform views. This will run a task on
+  // the UI thread before returning.
+  Shell& shell = Shell::Shared();
+  std::vector<base::WeakPtr<PlatformView>> platform_views;
+  shell.WaitForPlatformViews(&platform_views);
+
+  JSONResponseBuilder builder;
+  builder.Append("{\"type\":\"FlutterViewList\",\"views\":[");
+  bool prefix_comma = false;
+  for (auto it = platform_views.begin(); it != platform_views.end(); it++) {
+    PlatformView* view = it->get();
+    if (!view) {
+      // Skip any platform views which have been deleted.
+      continue;
+    }
+    if (prefix_comma) {
+      builder.Append(",");
+    } else {
+      prefix_comma = true;
+    }
+    builder.Append("{\"type\":\"FlutterView\", \"id\": \"_flutterView/");
```

Comment:
You can use the raw literal syntax to avoid having to escape these yourself http://en.cppreference.com/w/cpp/language/string_literal (item 6). This is fine too. Just a thought.